assets/src/data/model/entity-factory/create.js

```diff
@@ -0,0 +1,701 @@
+	const opts = { configurable: true };
+	if ( isArray( primaryKeys ) ) {
+		createGetter(
+			instance,
+			'primaryKey',
+			primaryKeys[ 0 ],
+			opts
+		);
+		createGetterAndSetter(
+			instance,
+			'primaryKeys',
+			primaryKeys,
+			opts
+		);
+		createCallbackGetter(
+			instance,
+			'hasMultiplePrimaryKeys',
+			hasMultiplePrimaryKeysCallback,
+			opts
+		);
+	}
+};
+
+/**
+ * @param {Object} instance
+ * @return {function(string): boolean} Returns a callback for the
+ * hasCalculatedField getter
+ */
+const hasCalculatedFieldCallback = ( instance ) =>
+	( fieldNameToCheck ) => ! isUndefined( instance[ fieldNameToCheck ] );
+
+/**
+ * Creates the getters for all the calculated fields and value on the entity.
+ * @param {Object} instance
+ * @param {Object.<string,*>}fieldsAndValues
+ */
+export const setCalculatedFieldAndValues = ( instance, fieldsAndValues ) => {
+	forEach( fieldsAndValues, ( calculatedFieldValue, calculatedFieldName ) => {
+		createGetter(
+			instance,
+			camelCase( calculatedFieldName ),
+			calculatedFieldValue
+		);
+	} );
+	createCallbackGetter(
+		instance,
+		'hasCalculatedField',
+		hasCalculatedFieldCallback
+	);
+};
+
+/**
+ * Create getters for the various resource links on the entity.
+ *
+ * @param {Object} instance
+ * @param {Object.<string,*>}fieldsAndValues
+ */
+export const setResources = ( instance, fieldsAndValues ) => {
+	const relations = [];
+	let relationName;
+	forEach( fieldsAndValues, ( resourceValue, resourceName ) => {
+		if ( resourceName === 'self' ) {
+			createGetter( instance, 'resourceLink', resourceValue[ 0 ].href );
+		} else if ( resourceName === 'collection' ) {
+			createGetter(
+				instance,
+				'collectionResourceLink',
+				resourceValue[ 0 ].href
+			);
+		} else {
+			relationName = getRelationNameFromLink( resourceName );
+			relations.push( relationName );
+			setRelationsResource(
+				instance,
+				relationName + 'Resource',
+				resourceValue
+			);
+		}
+	} );
+	//set relations getter
+	createGetter( instance, 'getRelations', relations );
+};
+
+/**
+ * @param {Object} instance
+ * @return {function(string): Object} Returns the callback for getting a
+ * relation resource
+ */
+const getRelationResourceCallback = ( instance ) =>
+	( relationName ) => instance[ relationName.replace( 'Resource', '' ) ];
+
+/**
+ * Creates getters for the relations resource object.
+ *
+ * @param {Object} instance
+ * @param {string} relationName
+ * @param {Object.<string, string>} resourceInfo
+ */
+export const setRelationsResource = (
+	instance,
+	relationName,
+	resourceInfo
+) => {
+	createGetter(
+		instance,
+		relationName,
+		{
+			resourceLink: resourceInfo[ 0 ].href,
+			single: resourceInfo[ 0 ].single,
+		}
+	);
+	if ( isUndefined( instance.getRelationResource ) ) {
+		createCallbackGetter( instance,
+			'getRelationResource',
+			getRelationResourceCallback
+		);
+	}
+};
+
+/**
+ * Sets the internal save state to the given value.
```

Comment:
> (I'm assuming dirty state is reset after updating via the API through some other mechanism such as generating a new object?)

Ya currently the plan is when a repo persists a dirty object it will replace it with a new entity object based on the response from the endpoint.

I'll update the comment block here.